pkg/operator/injector/injector.go

```diff
@@ -0,0 +1,487 @@
+					s.configmapVolumeMount}
+			}
+			if c.Env != nil {
+				pod.Spec.Containers[i].Env = append(pod.Spec.Containers[i].Env, s.env)
+			} else {
+				pod.Spec.Containers[i].Env = []corev1.EnvVar{s.env}
+			}
+			break
+		} else {
+			if c.VolumeMounts != nil {
+				pod.Spec.Containers[i].VolumeMounts = append(pod.Spec.Containers[i].VolumeMounts,
+					s.sidecarVolumeMount, s.configmapVolumeMount)
+			} else {
+				pod.Spec.Containers[i].VolumeMounts = []corev1.VolumeMount{s.sidecarVolumeMount,
+					s.configmapVolumeMount}
+			}
+			if c.Env != nil {
+				pod.Spec.Containers[i].Env = append(pod.Spec.Containers[i].Env, s.env)
+			} else {
+				pod.Spec.Containers[i].Env = []corev1.EnvVar{s.env}
+			}
+		}
+	}
+}
+
+// GetInjectStrategy gets user's injection strategy
+func (s *SidecarInjectField) GetInjectStrategy(containers []corev1.Container, labels,
+	annotation *map[string]string) {
+	// set default value
+	s.needInject = false
+	s.injectContainer = ""
+	s.agentOverlay = false
+
+	// set needInject's value , if the pod has the label "swck-java-agent-injected=true",means need inject
+	if *labels == nil {
+		labels = new(map[string]string)
+		s.needInject = false
+	}
+
+	switch strings.ToLower((*labels)[labelKeyagentInjector]) {
+	case "true":
+		s.needInject = true
+	}
+
+	// set injectContainer's value
+	if *annotation == nil {
+		annotation = new(map[string]string)
+		s.injectContainer = ""
+		s.agentOverlay = false
+	}
+
+	// validate the container is or not exist
+	if v, ok := (*annotation)[AnnoInjectContainerName.Name]; ok {
+		for _, c := range containers {
+			if c.Name == v {
+				s.injectContainer = v
+				break
+			}
+		}
+	}
+
+	// set agentOverlay's value
+	switch strings.ToLower((*annotation)[AnnoAgentConfigOverlay.Name]) {
+	case "true":
+		s.agentOverlay = true
+	}
+
+}
+
+func (s *SidecarInjectField) injectErrorAnnotation(annotation *map[string]string, errorInfo string) {
+	(*annotation)[AnnoInjectErrorInfo.Name] = errorInfo
+}
+
+// SidecarOverlayandGetValue get final value of sidecar
+func (s *SidecarInjectField) SidecarOverlayandGetValue(as *Annotations, annotation *map[string]string,
+	a Annotation) (string, bool) {
+	if _, ok := (*annotation)[a.Name]; ok {
+		err := as.SetOverlay(annotation, a)
+		if err != nil {
+			s.injectErrorAnnotation(annotation, err.Error())
+			return "", false
+		}
+	}
+	return as.GetFinalValue(a), true
+}
+
+func (s *SidecarInjectField) setValue(config *string, as *Annotations, annotation *map[string]string,
+	a Annotation) bool {
+	if v, ok := s.SidecarOverlayandGetValue(as, annotation, a); ok {
+		*config = v
+		return true
+	}
+	return false
+}
+
+// OverlaySidecar overlays default config
+func (s *SidecarInjectField) OverlaySidecar(as *Annotations, annotation *map[string]string) bool {
+	var (
+		command       string
+		argOption     string
+		argCommand    string
+		configmapName string
+	)
+
+	switch {
+	case !s.setValue(&s.initcontainer.Name, as, annotation, AnnoInitcontainerName):
```

Comment:
Too many branches here. A common way is to store annotations into a `slice`, loop which to validate them.